ADAL/src/urlprotocol/ADCustomHeaderHandler.m

```diff
@@ -20,42 +20,47 @@
 
 @implementation ADCustomHeaderHandler
 
-static NSDictionary* customHeaders;
-static NSMutableDictionary* customHeadersForSingleUse;
+static NSMutableDictionary* s_customHeaders = nil;
+static NSMutableDictionary* s_customHeadersForSingleUse = nil;
 
 + (void)initialize
 {
-    if (self == [ADCustomHeaderHandler class]) {
-        customHeaders = @{
-                         @"x-ms-PkeyAuth":@"1.0",
-                         };
-        customHeadersForSingleUse = [NSMutableDictionary new];
+    if (self == [ADCustomHeaderHandler class])
+    {
+        s_customHeaders =
+            [NSMutableDictionary dictionaryWithDictionary: @{@"x-ms-PkeyAuth":@"1.0"}];
+        SAFE_ARC_RETAIN(s_customHeaders);
+        s_customHeadersForSingleUse = [NSMutableDictionary new];
```

Comment:
do we need dealloc function in this class to release objects like s_customHeadersForSingleUse?